Builder/SolariumSelectQueryBuilder.php

```diff
@@ -111,96 +106,94 @@ public function buildSolariumQueryFromGeneric(GenericSelectQuery $query)
         }
 
         //if there are facets to request, request them
-        if ($this->hasSearchContext()) {
-            $facets = $this->getSearchContext()->getFacets();
+        $facets = $query->getFacets();
+        if (!empty($facets)) {
             $facetNamesToExclude = $query->getFacetNamesToExclude();
-            if (!empty($facets)) {
-                foreach ($facets as $facet) {
-                    //if query indicates we should skip this facet, then skip it
-                    if (false !== array_search($facet->getSearchKey(), $facetNamesToExclude)) {
-                        continue;
-                    }
-                    $solariumFacets = [];
-                    //check whether to request missing facet values
-                    $checkMissingFacetValues = (method_exists($this->getSearchContext(), 'shouldRequestFacetValueForMissing')) ? $this->getSearchContext()->shouldRequestFacetValueForMissing() : null;
```

Comment:
I added this method to the SearchContextInterface and repaired all implementations. Considered this line a bit sketchy :)